services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeMapper.kt

```diff
@@ -0,0 +1,84 @@
+package fi.hel.haitaton.hanke
+
+import fi.hel.haitaton.hanke.ContactType.MUU
+import fi.hel.haitaton.hanke.ContactType.OMISTAJA
+import fi.hel.haitaton.hanke.ContactType.RAKENNUTTAJA
+import fi.hel.haitaton.hanke.ContactType.TOTEUTTAJA
+import fi.hel.haitaton.hanke.domain.Hanke
+import fi.hel.haitaton.hanke.domain.HankeYhteystieto
+import fi.hel.haitaton.hanke.domain.Hankealue
+import fi.hel.haitaton.hanke.geometria.Geometriat
+import fi.hel.haitaton.hanke.tormaystarkastelu.TormaystarkasteluTulos
+
+object HankeMapper {
+
+    /**
+     * Needs to be called inside a transaction for lazy fields to be accessible.
+     *
+     * @param entity Source hanke entity.
+     * @param geometriaData Since hanke only has an id reference on the actual geometries.
+     */
+    fun domainFrom(entity: HankeEntity, geometriaData: Map<Int, Geometriat?>): Hanke =
+        with(entity) {
+            Hanke(
+                    id = id,
+                    hankeTunnus = hankeTunnus,
+                    onYKTHanke = onYKTHanke,
+                    nimi = nimi,
+                    kuvaus = kuvaus,
+                    vaihe = vaihe,
+                    suunnitteluVaihe = suunnitteluVaihe,
+                    version = version,
+                    createdBy = createdByUserId,
+                    createdAt = createdAt?.zonedDateTime(),
+                    modifiedBy = modifiedByUserId,
+                    modifiedAt = modifiedAt?.zonedDateTime(),
+                    status = status,
+                    generated = generated
+                )
+                .apply {
```

Comment:
Closing `with` here would make sense, so we don't have overlapping `this`s. Like, without checking it's hard to say if `tormaystarkasteluTulokset` is a part of the `Hanke` or the `HankeEntity`.
```suggestion
                )}
                .apply {
```

Reply:
Fixed.